Marlin/ultralcd.cpp

```diff
@@ -1083,6 +1092,51 @@ void lcd_cooldown() {
 
 /**
  *
+ * "Change Filament" submenu
+ *
+ */
+static void lcd_change_filament_menu() {
+  START_MENU();
+  MENU_ITEM(back, MSG_MAIN);
+  MENU_ITEM(function, MSG_REMOVE_FILAMENT , lcd_retract_filament);
+  MENU_ITEM(submenu, MSG_INSERT_FILAMENT , lcd_insert_filament_menu);
+  END_MENU();
+}
+
+static void lcd_insert_filament_menu() {
+  START_MENU();
+  MENU_ITEM(back, MSG_CHANGE_FILAMENT);
+  MENU_ITEM_EDIT_CALLBACK(bool, MSG_INSERT_FILAMENT_SLOW ,&filamentSlowInsert ,lcd_insert_filament_slow);
+  if(!filamentSlowInsert)
+    MENU_ITEM(function, MSG_INSERT_FILAMENT  , lcd_insert_filament);
+  END_MENU();
+}
+
+static void lcd_insert_filament_slow(){
+ while (filamentSlowInsert) {
```

Comment:
After mulling it over, I think recursing into the menus isn't so bad. It eats 2 bytes of stack space, but otherwise it's not problematic. The only danger is accidentally making some code that could keep recursing and blow the stack. Not a problem with your code here at all.

I did a patch-up of this PR to illustrate the "Marlin way" to put together all the bits that you're bringing together. Check out #3605 and see if it makes sense to you how I've reworked the code to be more in the idiom that (for better or worse) has been established for lcd menus and optional features.